From e9aef13188593ebf06b9dc2d7de5d664c9790d6a Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Thu, 3 Jun 2021 19:23:13 +0100
Subject: [PATCH] build: Make the directory for helper executables consistent
 with Autotools

The Autotools build system has been using /usr/lib/polkit-1 for several
releases, even on distributions where the library directory is /usr/lib64
or /usr/lib/x86_64-linux-gnu, so it makes sense for Meson to do the same.
This lets 32- and 64-bit polkit agents share a single helper executable.

This might be superseded by polkit!63, which requests going back to using
the libexecdir for these (like polkit 0.105 did), which would also make
sense; but until that's decided, let's at least be consistent between
our two build systems.

Every time we change this, all programs that have already loaded
libpolkit-agent into their address space need to be restarted, unless
distributions provide compatibility symlinks.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 data/meson.build              | 2 +-
 meson.build                   | 5 +++--
 src/polkitagent/meson.build   | 2 +-
 src/polkitbackend/meson.build | 2 +-
 4 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 14c100b..cad7ccd 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,4 +1,4 @@
-service_conf = {'libprivdir': pk_prefix / pk_pkglibdir}
+service_conf = {'libprivdir': pk_prefix / pk_libprivdir}
 
 configure_file(
   input: 'org.freedesktop.PolicyKit1.service.in',
diff --git a/meson.build b/meson.build
index bd4ec6b..81715b5 100644
--- a/meson.build
+++ b/meson.build
@@ -27,7 +27,8 @@ pk_sysconfdir = get_option('sysconfdir')
 
 pk_pkgdatadir = pk_datadir / pk_api_name
 pk_pkgincludedir = pk_includedir / pk_api_name
-pk_pkglibdir = pk_libdir / pk_api_name
+# note that this is always 'lib', not lib64 or lib/x86_64-linux-gnu
+pk_libprivdir = 'lib' / pk_api_name
 pk_pkgsysconfdir = pk_sysconfdir / pk_api_name
 
 pk_actiondir = pk_api_name / 'actions'
@@ -324,7 +325,7 @@ meson.add_install_script(
   'meson_post_install.py',
   get_option('bindir'),
   pk_pkgdatadir,
-  pk_pkglibdir,
+  pk_libprivdir,
   pk_pkgsysconfdir,
   polkitd_user,
 )
diff --git a/src/polkitagent/meson.build b/src/polkitagent/meson.build
index 84aa260..bee3820 100644
--- a/src/polkitagent/meson.build
+++ b/src/polkitagent/meson.build
@@ -115,5 +115,5 @@ executable(
   dependencies: deps,
   c_args: '-D_POLKIT_COMPILATION',
   install: true,
-  install_dir: pk_pkglibdir,
+  install_dir: pk_libprivdir,
 )
diff --git a/src/polkitbackend/meson.build b/src/polkitbackend/meson.build
index 93c3c34..64f0e4a 100644
--- a/src/polkitbackend/meson.build
+++ b/src/polkitbackend/meson.build
@@ -68,5 +68,5 @@ executable(
   c_args: c_flags,
   link_with: libpolkit_backend,
   install: true,
-  install_dir: pk_pkglibdir,
+  install_dir: pk_libprivdir,
 )
-- 
2.33.0.rc2

